### ISSUE TITLE
[new-cap] false positive when constructor is returned by chained function call

### ISSUE THREAD

**Issue by jrencz** [eslint/eslint#4288]
Please consider the following snippet:

```es6
const foo = {bar: function BazService () {}};

const namesapceCahce = new Map();
namesapceCahce.set("bar", foo.bar);

const bazInstance1 = new namesapceCahce.get("bar").BazService();
const bazInstance2 = new foo.bar.BazService();

bazInstance1("quux");
bazInstance2("quux");
```

When pasted into [demo](http://eslint.org/demo/) with default settings + `es6` and `new-cap` it produces 2 errors for line 6 which uses Map#get to access the constructor previously stored in `serviceCache`:
```
6:41 - A constructor name should not start with a lowercase letter. (new-cap)
6:52 - A function with a name starting with an uppercase letter should only be used as a constructor. (new-cap)
```

Note that in line 7, where the same constructor is accessed, but without function call in chain there's no error reported.

Expected behaviour for `new-cap` in this case is not to report any error.

esLint 1.7.3

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by nzakas** [comment]
Thanks, we will take a look.

**Comment by platinumazure** [comment]
Is this really a false positive? The `new` operator only parses until the
first set of parentheses, so the actual constructor invocation is a
non-constructor function call.

**Comment by jrencz** [comment]
I found that wrapping the whole constructor with parentheses like this:

```js
const bazInstance1 = new (namesapceCache.get("bar").BazService)();
```

prevents the error.
And in fact that snippet was not working at all. Parentheses are required in such case.
Closing
